qt-models/weightmodel.cpp

```diff
@@ -102,7 +102,7 @@ bool WeightModel::setData(const QModelIndex &index, const QVariant &value, int r
 			if (!ws->description || gettextFromC::tr(ws->description) != vString) {
 				// loop over translations to see if one matches
 				int i = -1;
-				while (i < MAX_WS_INFO && ws_info[++i].name) {
+				while (i < MAX_WS_INFO - 1 && ws_info[++i].name) {
```

Comment:
My guess is that this will change once we no longer have a fixed array of weight systems

Reply:
Careful: this is not an array of weightsystems, but an array of weightsystem infos. The fixed array of weightsystems is long gone. :)

But yes, that loop is horrible, and if I remember correctly, I removed it in #2363. Which makes me wonder if it is worth generating a merge-conflict here.

And which reminds me that I have to look at #2363 again...

Reply:
I am happy to leave it as it is. Even if we access out of bounds, that is only read access, so no danger of overwriting other data.

Reply:
It felt wrong when I wrote my comment above. Urgs. Sorry @bstoeger 

Yes, let's drop this one and assume that #2363 will take care of it.